### PR TITLE
Improve global search

### PR DESCRIPTION
**Pull request by JaiZed**
Used Levenshtein distance to sort by closest match.
Increased total limit to 30.
Increased dropdown height to show more results initially (and which can also be scrolled into view).
After upgrade to next Mantine major version, scrollbars will appear automatically (when that code is uncommented).

### CODE REVIEW

**Review by anderson-oki — APPROVED** [review]

File: frontend/src/components/Search.tsx [hunk 1]
```diff
@@ -85,6 +85,10 @@ const Search: FunctionComponent = () => {
       size="sm"
       data={results}
       value={query}
+      limit={30}
+      // uncomment following line after upgrade to Mantine 7.x or higher
```
Comment (anderson-oki) [inline frontend/src/components/Search.tsx:89]: Can you prefix this with TODO:

```suggestion
      // TODO: uncomment following line after upgrade to Mantine 7.x or higher
```

**Comment by JaiZed** [comment]
I was just going to wait and remove the comment next week when the Mantine upgrade was done.

**Comment by morpheus65535** [comment]
Levenshtein module isn't part of Bazarr and isn't part of base Python install. This wont work unless you add it to vendored libs. You want me to do it? I haven't created a script or a wiki on how to do this yet.

**Comment by morpheus65535** [comment]
Levenshtein rely on cmake that need to be compiled on destination computer. We can't vendor this module. If it's explicitly required, we'll need to add it to requirements.txt and make sure it will be available or can be compiled on all major platforms.

Does it worth the effort?

**Comment by JaiZed** [comment]
I think it's worth the effort because the sorted results are so much better.
However, I will look to see if there are any other options that don't require this module.

I'm wondering how it got into my Python environment. Perhaps a side effect on installing whisper or something else?

**Comment by JaiZed** [comment]
I have found a Python library called textdistance that has the same functionality.
`pip install textdistance`

It seems to work the same and I think it is pure Python.
Is that acceptable?

**Comment by morpheus65535** [comment]
> Is that acceptable?

Yes absolutely. I've added it to your PR.

File: frontend/src/components/Search.tsx [hunk 2]
```diff
@@ -85,6 +85,10 @@ const Search: FunctionComponent = () => {
       size="sm"
       data={results}
       value={query}
+      limit={30}
+      // TODO: uncomment following line after upgrade to Mantine 7.x or higher
+      // scrollAreaProps = {{type: auto}}
+      maxDropdownHeight={400}
```
Comment (anderson-oki) [inline frontend/src/components/Search.tsx:91]: @JaiZed 

I dare to say you don't need this after Mantine is updated to 7. I will leave you to check again without this once the mantine 7 is merged.

**Comment by morpheus65535** [comment]
mantine 7 has been merged. Anything to fine tune on this PR?

**Pull request closed without merging**